forge-ai/src/main/java/forge/ai/AiController.java

```diff
@@ -1718,6 +1718,16 @@ public final boolean aiShouldRun(final ReplacementEffect effect, final SpellAbil
             hostCard = game.getCardState(hostCard);
         }
 
+        if (effect.hasParam("CommanderMoveReplacement")) {
```

Comment:
what is this for - 
it should already reach the logic in `ChangeZoneAi.doReturnCommanderLogic`?

Reply:
Ah i didn't know about that function. But what I'm adding is still needed. Bouncing to hand shouldn't send to Command Zone.

Reply:
yea most likely only in corner cases
might still be good to move it to the same place? 🤔

Reply:
Yep, I'll move it to the function you pointed out. THanks for the eyes.